### PR TITLE
Create Networking class and Logger

### PR DESCRIPTION
**Pull request by callum-code**
This is personally how I like to do networking. 

Usage: 
```
Network.post(
    url: "example.com/uploads",
    data: [
        "id" : "q938ur83r",
        "image" : bytes
    ],
    header: [ "Authorization" : "Bearer \(token)" ],
    contentType: .json,
    completionHandler: { error, data in
        // handle result
    }
)
```

### CODE REVIEW

File: iOS/MMIWG2S/Utilities/Network.swift [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+//
+//  Created by Callum Johnston on 4/7/21.
+//  Copyright © 2021 Google LLC. All rights reserved.
+//
+
+import Foundation
+
+enum HttpContentType: String {
+    case json = "application/json"
+    case form = "application/x-www-form-urlencoded"
```
Comment (pavan4) [inline iOS/MMIWG2S/Utilities/Network.swift:13]: At this point the server is not setup to support `x-www-form-urlencoded`. It supports a `multipart/form-data` for accepting the image file uploaded. This case needs to be added here. 

Adding files to a form request requires setting the `boundaries` and also the `MimeType`
Here is a snippet for handling `multipart/form-data` in `iOS`: [link](https://stackoverflow.com/a/64642247/1072825)

File: iOS/MMIWG2S/Utilities/Network.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    case all = "*/*"
+}
+
+class Network {
+    
+    /**
+     - Parameter url: URL for the POST request
+     - Parameter data: the data to be posted
```
Comment (Mehequanna) [inline iOS/MMIWG2S/Utilities/Network.swift:21]: Will this data handle an image/file as well?

**Review by Mehequanna — APPROVED** [review]
I may need you to show me how to add an image file to the Post, but other than that looks great! Thanks!

File: iOS/MMIWG2S/Utilities/Network.swift (hunk 2 above)
Comment (callum-code) [reply iOS/MMIWG2S/Utilities/Network.swift:21]: absolutely! We can just pass in the image as a `Data` representation of the bytes.

File: iOS/MMIWG2S/Utilities/Network.swift (hunk 1 above)
Comment (callum-code) [reply iOS/MMIWG2S/Utilities/Network.swift:13]: Oh! Good to know, thanks @pavan4!

**Comment by callum-code** [comment]
How's this @pavan4?
```
let image = UIImage(named: "face-background")
let metadataJson = [
    "name" : "callum",
    "city" : "Portland"
]
guard let metadata = try? JSONSerialization.data(withJSONObject: metadataJson, options: .prettyPrinted),
      let imageData = image?.jpegData(compressionQuality: 1.0)
else { return }
Network.post(
    url: "example.com/uploads",
    data: [
        "metadata" : ("application/json", metadata),
        "file" : ("image/jpeg", imageData)
    ],
    header: [ "Authentication" : "Bearer: \(whatever)" ],
    contentType: .multipartForm,
    completionHandler: { error, response in
        // handle response
    }
)
```

**Comment by Mehequanna** [comment]
I have a commit on this branch to test this code: `stephen/testing-callum-network`
Here's the code I'm using. I'm currently getting a 500 error, I'll list below the code. I'll defer to Pavan on whether it's being called correctly.
```
        let image = imageToShare.compress(to: 1000)
        let metadataJson = [
            "name" : "iOS Test",
            "email" : "PortlandTest@test.com"
        ]
        guard let metadata = try? JSONSerialization.data(withJSONObject: metadataJson, options: .prettyPrinted),
              let imageData = image.pngData()
        else { return }
        Network.post(
            url: "https://mmiw.mehequanna.com/submissions",
            data: [
                "metadata" : ("application/json", metadata),
                "file" : ("image/png", imageData)
            ],
            header: [ "Authentication" : "Bearer: Not currently used." ],
            contentType: .multipartForm,
            completionHandler: { error, response in
                
                print("Something went wrong \(String(describing: error))?")
                print("Something went right \(String(describing: response))?")
                // handle response
            }
        )
```


500 Error
```
🛑 HTTP ERROR: 500: <NSHTTPURLResponse: 0x282727500> { URL: https://mmiw.mehequanna.com/submissions } { Status Code: 500, Headers {
    "Content-Length" =     (
        42
    );
    "Content-Type" =     (
        "application/json; charset=utf-8"
    );
    Date =     (
        "Fri, 09 Apr 2021 04:34:01 GMT"
    );
    Etag =     (
        "W/\"2a-GsHqaf0kMfGHw0++t8qX12ZBNII\""
    );
    Server =     (
        nginx
    );
    "Strict-Transport-Security" =     (
        "max-age=15552000; includeSubDomains",
        "max-age=31536000"
    );
    "x-content-type-options" =     (
        nosniff
    );
    "x-dns-prefetch-control" =     (
        off
    );
    "x-download-options" =     (
        noopen
    );
    "x-frame-options" =     (
        SAMEORIGIN
    );
    "x-ratelimit-limit" =     (
        10
    );
    "x-ratelimit-remaining" =     (
        9
    );
    "x-ratelimit-reset" =     (
        1617954085
    );
    "x-xss-protection" =     (
        "1; mode=block"
    );
} }
```

**Comment by callum-code** [comment]
Hmmm I'm not sure that request would work. Does the request require any type of authorization? If not then you definitely won't want that authentication header. Are `metadata` and `file` the correct names? Those were just examples I was using. Could you give an example of a working request using `curl -v -F`?

**Comment by Mehequanna** [comment]
Oh you're right. I'll try another pass tomorrow. Here's the curl that works:

curl -F "name=Test" \
-F "email=MMIW1@gmail.com" \
-F "image=@/Users/stephenemery/Downloads/asia.jpg" \                          
-F "verified=1" \
https://mmiw.mehequanna.com/submissions

**Comment by callum-code** [comment]
> Oh you're right. I'll try another pass tomorrow. Here's the curl that works:
> 
> curl -F "name=Test" 
> -F "email=[MMIW1@gmail.com](mailto:MMIW1@gmail.com)" 
> -F "image=@/Users/stephenemery/Downloads/asia.jpg" \
> -F "verified=1" 
> https://mmiw.mehequanna.com/submissions

Ah I see. Ok I made some changes to meet that exact format, but still I'm getting a `500` error. @pavan4 any thoughts? Sample request:
```
let image = UIImage(named: "hand-background")
guard let imageData = image?.pngData() else { return }
Network.post(
    url: "https://mmiw.mehequanna.com/submissions",
    metadata: [
        "name" : "test",
        "email" : "test@email.com",
        "verified" : "1"
    ],
    data: [
        "image" : ("image/png", imageData)
    ],
    contentType: .multipartForm,
    completionHandler: { error, response in
        
    }
)
```

**Comment by pavan4** [comment]
Hi @callum-code 

I forked your branch and created a PR with a fix for this with https://github.com/nativesintech/MMIWG2S/pull/77 

You can call this easily like so, 

```
Network.post(
    url: "http://192.168.1.148:3000/submissions",
    data: [
        "name" : ("text", "iOS Test".data(using: .utf8)),
        "email" : ("text", "PortlandTest@test.com".data(using: .utf8)),
        "image" : ("image/png", imageData)
    ],
    contentType: .multipartForm,
    completionHandler: { error, response in

    }
)
```

**Pull request merged**